sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/generated/OpenAIClientTestBase.java

```diff
@@ -4,40 +4,127 @@
 import com.azure.core.http.HttpClient;
 import com.azure.core.http.policy.HttpLogDetailLevel;
 import com.azure.core.http.policy.HttpLogOptions;
 import com.azure.core.test.TestBase;
 import com.azure.core.test.TestMode;
+import com.azure.core.test.TestProxyTestBase;
 import com.azure.core.util.Configuration;
 import com.azure.identity.DefaultAzureCredentialBuilder;
 import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.junit.jupiter.api.Test;
 import reactor.core.publisher.Mono;
 
-class OpenAIClientTestBase extends TestBase {
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+abstract class OpenAIClientTestBase extends TestProxyTestBase {
     protected OpenAIClient openAIClient;
+    protected OpenAIAsyncClient openAIAsyncClient;
 
     @Override
     protected void beforeTest() {
         OpenAIClientBuilder openAIClientbuilder =
                 new OpenAIClientBuilder()
-                        .endpoint(Configuration.getGlobalConfiguration().get("ENDPOINT", "endpoint"))
+                        .endpoint(Configuration.getGlobalConfiguration().get("AZURE_OPENAI_ENDPOINT", "endpoint"))
                         .httpClient(HttpClient.createDefault())
                         .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC));
         if (getTestMode() == TestMode.PLAYBACK) {
             openAIClientbuilder
-                    .httpClient(interceptorManager.getPlaybackClient())
-                    .credential(request -> Mono.just(new AccessToken("this_is_a_token", OffsetDateTime.MAX)));
+                .httpClient(interceptorManager.getPlaybackClient())
+                .credential(request -> Mono.just(new AccessToken("this_is_a_token", OffsetDateTime.MAX)));
         } else if (getTestMode() == TestMode.RECORD) {
             openAIClientbuilder
-                    .addPolicy(interceptorManager.getRecordPolicy())
-                    .credential(new DefaultAzureCredentialBuilder().build());
+                .addPolicy(interceptorManager.getRecordPolicy())
+                .credential(new AzureKeyCredential(Configuration.getGlobalConfiguration().get("AZURE_OPENAI_KEY")));
+//                    .credential(new DefaultAzureCredentialBuilder().build()); // Uncomment it if using AAD
         } else if (getTestMode() == TestMode.LIVE) {
             openAIClientbuilder.credential(new DefaultAzureCredentialBuilder().build());
         }
         openAIClient = openAIClientbuilder.buildClient();
+        openAIAsyncClient = openAIClientbuilder.buildAsyncClient();
+    }
+
+    @Test
+    public abstract void getCompletions(HttpClient httpClient, OpenAIServiceVersion serviceVersion);
```

Comment:
I don't think I see any concrete test written, but should we already see a `session-records` folder with recorded json for tests?

Reply:
Yeah. I just add the recorded JSON files